shell.nix

```diff
@@ -0,0 +1,26 @@
+let
+  mozillaOverlay =
+    import (builtins.fetchGit {
+      url = "https://github.com/mozilla/nixpkgs-mozilla.git";
+      rev = "78e723925daf5c9e8d0a1837ec27059e61649cb6";
+    });
+  nixpkgs = import <nixpkgs> { overlays = [ mozillaOverlay ]; };
+  rust-nightly = with nixpkgs; ((rustChannelOf { date = "2022-11-15"; channel = "nightly"; }).rust.override {
```

Comment:
We already have rust-toolchain.toml to control the version. Any specific version here will be overriden.

Reply:
I use nixos for dev, just in the nix way.

Reply:
I mean, `cargo` will override this. Unless you use `rustc` to compile the project instead of `cargo`. Otherwise, this version is useless.

Reply:
Nixpkg is used for a package manager, there is no need for another package manager to deal with it.